p2p/starknet/starknet_test.go

```diff
@@ -179,15 +174,100 @@ func TestClientHandler(t *testing.T) {
 	})
 
 	t.Run("get txns", func(t *testing.T) {
-		res, cErr := client.RequestTransactions(testCtx, &spec.TransactionsRequest{})
+		blocks := []*core.Block{
+			{
+				Header: &core.Header{
+					Number: 0,
+				},
+				Transactions: []core.Transaction{
+					fillFelts(t, &core.DeployTransaction{
+						ConstructorCallData: feltSlice(3),
+					}),
+					fillFelts(t, &core.L1HandlerTransaction{
+						CallData: feltSlice(2),
+						Version:  txVersion(1),
+					}),
+				},
+			},
+			{
+				Header: &core.Header{
+					Number: 1,
+				},
+				Transactions: []core.Transaction{
+					fillFelts(t, &core.DeployAccountTransaction{
+						DeployTransaction: core.DeployTransaction{
+							ConstructorCallData: feltSlice(3),
+							Version:             txVersion(1),
+						},
+						TransactionSignature: feltSlice(2),
+					}),
+				},
+			},
+			{
+				Header: &core.Header{
+					Number: 2,
+				},
+				Transactions: []core.Transaction{
+					fillFelts(t, &core.DeclareTransaction{
+						TransactionSignature: feltSlice(2),
+						Version:              txVersion(0),
+					}),
+					fillFelts(t, &core.DeclareTransaction{
+						TransactionSignature: feltSlice(2),
+						Version:              txVersion(1),
+					}),
+				},
+			},
+			{
+				Header: &core.Header{
+					Number: 3,
+				},
+				Transactions: []core.Transaction{
+					fillFelts(t, &core.InvokeTransaction{
+						CallData:             feltSlice(3),
+						TransactionSignature: feltSlice(2),
+						Version:              txVersion(0),
+					}),
+					fillFelts(t, &core.InvokeTransaction{
+						CallData:             feltSlice(4),
+						TransactionSignature: feltSlice(2),
+						Version:              txVersion(1),
+					}),
+				},
+			},
+		}
+		numOfBlocks := uint64(len(blocks))
+
+		for _, block := range blocks {
+			mockReader.EXPECT().BlockByNumber(block.Number).Return(block, nil)
+		}
+
+		res, cErr := client.RequestTransactions(testCtx, &spec.TransactionsRequest{
+			Iteration: &spec.Iteration{
+				Start: &spec.Iteration_BlockNumber{
+					BlockNumber: blocks[0].Number,
+				},
+				Direction: spec.Iteration_Forward,
+				Limit:     numOfBlocks,
+				Step:      1,
+			},
+		})
 		require.NoError(t, cErr)
 
-		count := uint64(0)
+		var count uint64
 		for txn, valid := res(); valid; txn, valid = res() {
+			if count == numOfBlocks {
+				assert.NotNil(t, txn.GetFin())
```

Comment:
I think we need to add a testcase to make sure when multiple blocks's transactions are requested then the fin has been sent correctly. However, that can be done in your other pr.